app/models/lookup_value.rb

```diff
@@ -70,7 +70,7 @@ def sanitize_match
   end
 
   def validate_and_cast_value
-    return if !self.value.is_a?(String) || value.contains_erb?
+    return if !self.value.is_a?(String) || value.contains_erb? == 0 || (value.contains_erb? && !lookup_key.merge_overrides)
```

Comment:
`value.contains_erb? == 0` only if it starts with erb, is this intentional?

Reply:
isn't `(value.contains_erb? && !lookup_key.merge_overrides)` already covered by `value.contains_erb?`?

Reply:
@tbrisker Yes, it should only return and thus not cast the value if the ERB is right at the beginning. If the ERB is further down in the YAML structure of the value, it should cast it to whatever value the user defines for the parameter.

Reply:
@orrabin Not exactly. Because I check for `value.contains_erb? == 0` all cases where the ERB is not in the beginning are not covered. But because I want to keep the impact of this fix as small as possible I also check if the parameter has the override flag set (because the bug doesn't show up otherwise).

Reply:
@FrediWeber I meant why did you split it to `==0` and add a condition that is already contained but I see your answer to @tbrisker about that. Thanks for clarifying.